### PR TITLE
[UI] Back from a detail page & update syntax highlighting in docs

### PR DESCRIPTION
**Pull request by lailahgrant**
Fixes https://github.com/taskcluster/taskcluster/issues/1558.

### CODE REVIEW

File: ui/src/views/Clients/ViewClient/index.jsx [hunk 1]
```diff
@@ -215,7 +215,7 @@ export default class ViewClient extends Component {
       }
 
       if (isNewClient) {
-        this.props.history.push({
+        this.props.history.replace({
```
Comment (helfi92) [inline ui/src/views/Clients/ViewClient/index.jsx]: 👍 This one looks good. The others should all have `.push`. In other words, the final diff should be:

```diff
diff --git a/ui/src/views/Clients/ViewClient/index.jsx b/ui/src/views/Clients/ViewClient/index.jsx
index 64ab92e47..5a5b853da 100644
--- a/ui/src/views/Clients/ViewClient/index.jsx
+++ b/ui/src/views/Clients/ViewClient/index.jsx
@@ -215,7 +215,7 @@ export default class ViewClient extends Component {
       }
 
       if (isNewClient) {
-        this.props.history.push({
+        this.props.history.replace({
           pathname: `/auth/clients/${encodeURIComponent(clientId)}`,
           state: { accessToken: result.data.createClient.accessToken },
         });
```

**Comment by helfi92** [comment]
It would be a good idea to rename the pull-request to be something more descriptive.

File: ui/src/views/Clients/ViewClient/index.jsx (hunk 1 above)
Comment (lailahgrant) [reply ui/src/views/Clients/ViewClient/index.jsx]: > 👍 This one looks good. The others should all have `.push`. In other words, the final diff should be:
> 
> ```diff
> diff --git a/ui/src/views/Clients/ViewClient/index.jsx b/ui/src/views/Clients/ViewClient/index.jsx
> index 64ab92e47..5a5b853da 100644
> --- a/ui/src/views/Clients/ViewClient/index.jsx
> +++ b/ui/src/views/Clients/ViewClient/index.jsx
> @@ -215,7 +215,7 @@ export default class ViewClient extends Component {
>        }
>  
>        if (isNewClient) {
> -        this.props.history.push({
> +        this.props.history.replace({
>            pathname: `/auth/clients/${encodeURIComponent(clientId)}`,
>            state: { accessToken: result.data.createClient.accessToken },
>          });
> ```


do you mean to say, i only change one line of push() to replace() and leave the other lines with push() ?

Comment (helfi92) [reply ui/src/views/Clients/ViewClient/index.jsx]: Correct.

**Comment by lailahgrant** [comment]
I have put push() in the  ui/src/views/Clients/ViewClient/index.jsx  on line 59...  is it right?

File: ui/src/views/Clients/ViewClients/index.jsx [hunk 2]
```diff
@@ -56,14 +56,14 @@ export default class ViewClients extends PureComponent {
...
-      this.props.history.push(
+      this.props.history.replace(
         search.length > 0 ? `?${stringify({ search })}` : '/auth/clients'
       );
     }
   };
 
   handleCreate = () => {
-    this.props.history.push('/auth/clients/create');
+    this.props.history.replace('/auth/clients/create');
```
Comment (helfi92) [inline ui/src/views/Clients/ViewClients/index.jsx]: This needs to stay as `push`.

File: ui/src/views/Clients/ViewClients/index.jsx [hunk 3]
```diff
@@ -56,14 +56,14 @@ export default class ViewClients extends PureComponent {
     });
 
     if (search !== searchUri) {
-      this.props.history.push(
+      this.props.history.replace(
```
Comment (helfi92) [inline ui/src/views/Clients/ViewClients/index.jsx]: This needs to stay as `push`.

File: ui/src/views/Clients/ViewClient/index.jsx [hunk 4]
```diff
@@ -102,7 +102,7 @@ export default class ViewClient extends Component {
   };
 
   handleDialogActionComplete = () => {
-    this.props.history.push(`/auth/clients`);
+    this.props.history.replace(`/auth/clients`);
```
[the remark this answers is not in the document]
Comment (lailahgrant) [reply ui/src/views/Clients/ViewClient/index.jsx]: ok sir, I have changed it.

File: ui/src/views/Clients/ViewClients/index.jsx (hunk 3 above)
Comment (helfi92) [reply ui/src/views/Clients/ViewClients/index.jsx]: No changes needed in `ViewClients.js`.

Comment (lailahgrant) [reply ui/src/views/Clients/ViewClients/index.jsx]: ok sir, thanks for the help... i have put it all the way you have directed me. Thanks sir.

Comment (helfi92) [reply ui/src/views/Clients/ViewClients/index.jsx]: Np. You have to push your changes for me to see them.

Comment (lailahgrant) [reply ui/src/views/Clients/ViewClients/index.jsx]: I have pushed them sir.

Comment (helfi92) [reply ui/src/views/Clients/ViewClients/index.jsx]: Awesome. The last thing to do is to add a changelog. From the root directory (not the ui directory), run `yarn changelog --silent --issue 1558`, then push the changes.

Comment (lailahgrant) [reply ui/src/views/Clients/ViewClients/index.jsx]: Sir, I have pushed it.. Thanks

**Comment by lailahgrant** [comment]
Ok Sir, and thank you

**Comment by lailahgrant** [comment]
Sir, is it possible for me to get an issue to work on??
thanks sir.

**Comment by helfi92** [comment]
Sure, assigned you https://github.com/taskcluster/taskcluster/issues/1772.

**Comment by helfi92** [comment]
@all-contributors please add @lailahgrant for code.

**Comment by helfi92** [comment]
For the future, it would be better to tackle issues in a separate PR instead of using the same branch :)

**Comment by lailahgrant** [comment]
Ok Sir.. 
Thanks.

File: ui/docs/manual/design/apis/hawk/temporary-credentials.mdx [hunk 5]
```diff
@@ -87,7 +87,7 @@ As described above, this can be any id that the issuer is permitted to create.
 
 In the certificate example from previous section, the string to sign would be:
 
-```
+```yaml
```
Comment (helfi92) [inline ui/docs/manual/design/apis/hawk/temporary-credentials.mdx]: I'm not sure this is YAML. We could probably just leave it alone.

File: ui/docs/manual/design/apis/hawk/temporary-credentials.mdx [hunk 6]
```diff
@@ -127,7 +127,7 @@ In this case, the certificate does not include an `issuer` property, and the
 `clientId` and `issuer` lines are omitted from the string used to generate the
 signature:
 
-```
+```yaml
```
Comment (helfi92) [inline ui/docs/manual/design/apis/hawk/temporary-credentials.mdx]: Same here. Let's not touch this.

File: ui/docs/reference/platform/auth/roles.mdx [hunk 7]
```diff
@@ -16,7 +16,7 @@ until no further expansion is possible, however, roles must be acyclic.
 
 Given the roles
 
-```
+```yaml
```
Comment (helfi92) [inline ui/docs/reference/platform/auth/roles.mdx]: This is not yaml. Let's not change it :)

File: ui/docs/tutorial/authenticate.mdx [hunk 8]
```diff
@@ -29,7 +29,7 @@ When you create the client, the site will give you an accessToken.  Copy that
 and set it in your shell session, along with the clientId you chose.  Sometihng
 like this:
 
-```
+```js
```
Comment (helfi92) [inline ui/docs/tutorial/authenticate.mdx]: ` ```bash `

File: ui/docs/tutorial/authenticate.mdx [hunk 9]
```diff
@@ -55,7 +55,7 @@ Credentials output as environment variables
 
 You can test the results with the same tool:
 
-```
+```json
```
Comment (helfi92) [inline ui/docs/tutorial/authenticate.mdx]: ` ```bash `

Comment (lailahgrant) [reply ui/docs/tutorial/authenticate.mdx]: Sir, I'm done changing them.
thanks.

Comment (helfi92) [reply ui/docs/tutorial/authenticate.mdx]: Don't forget to push your changes so that I am able to see them.

Comment (lailahgrant) [reply ui/docs/tutorial/authenticate.mdx]: i have pushed them Sir

**Pull request merged**